README.md

```diff
@@ -9,44 +9,63 @@
   <h1>Python to PlantUML</h1>
 </div>
 
-Generate Plantuml diagrams to document your python code
+Generate PlantUML class diagrams to document the business domain of your Python code.
 
 # How it works
 
+Using code [inspection](https://docs.python.org/3/library/inspect.html) (also called *reflexion* in other programming languages) and abstract tree parsing, `py2puml` produces a class diagram of your business code:
+* using the [PlantUML syntax](https://plantuml.com/en/class-diagram)
+* focuses on the business domain of your application
+* focuses on data structures: static and instance attributes, composition and inheritance relationships
+
 ## Features
 
-From a given path corresponding to a folder containing python code, `py2puml` loads each file as a module and generate a class diagram with the [PlantUML](https://plantuml.com/en/class-diagram) using:
+From a given path corresponding to a folder containing Python code, `py2puml` loads each file as a module and generates a class diagram of its defined datastructures with the [PlantUML](https://plantuml.com/en/class-diagram) using:
+
+* **[inspection](https://docs.python.org/3/library/inspect.html)** and [type annotations](https://docs.python.org/3/library/typing.html) to detect:
+  * static class attributes and [dataclass](https://docs.python.org/3/library/dataclasses.html) fields
+  * fields of [namedtuples](https://docs.python.org/3/library/collections.html#collections.namedtuple)
+  * members of [enumerations](https://docs.python.org/3/library/enum.html)
+  * composition and inheritance relationships (between your domain classes only, for documentation sake)
+
+* parsing [abstract syntax trees](https://docs.python.org/3/library/ast.html#ast.NodeVisitor) to detect the instance attributes defined in `__init__` constructors
 
-* inspection to detect the classes to document (see the [inspect](https://docs.python.org/3/library/inspect.html) module)
-* annotations (the python type hinting syntax) to detect the attributes and their types (see the [typing](https://docs.python.org/3/library/typing.html) module)
-* fields for classes derived from namedtuples
-* composition and inheritance relationships are drawn only between the domain classes (this is designed on purpose, for documentation sake)
+`py2puml` outputs diagrams in PlantUML syntax, which can be saved in text files along your python code and versioned with them.
```

Comment:
> Wouldn't it makes sense not to advocate for not versioning scripts

I think there is a double negation in this part but I got the meaning of the suggestion and I agree with the single-source-of-truth approach which would recommend to version code without what can be automatically derived from it.

However, here is what I came up doing in projects where I have no place to host some documentation:
1. I generate a PlantUML script with py2puml and version it so that new developers or project managers can view it online (in github / gitlab / bitbucket, which has a plugin to render it)
2. I ensure its consistency with a unit test which generates the class diagram with the Python API of py2puml and fails if the 2 PlantUML scripts do not match

Here is an example of such a "documentation unit-test": https://github.com/lucsorel/py2puml/blob/7-detect_instance_variables_in_constructor/tests/py2puml/test_py2puml.py#L10-L19

This is handy, but maybe not a practice to advocate :/

What do you think of this reformulation?
 `py2puml` outputs diagrams in PlantUML syntax, which can be hosted along other code documentation or maintained with a unit test (see [the `test_py2puml_model_on_py2uml_domain` test example](tests/py2puml/test_py2puml.py).).

Reply:
> I think there is a double negation in this part

Indeed, there's a spurious double negation.

I understand your use-case and can relate to it (online previews are useful). 

> What do you think of this reformulation?
> 
> > py2puml outputs diagrams in PlantUML syntax, which can be hosted along other code documentation or maintained with a unit test (see the test_py2puml_model_on_py2uml_domain test example.).
> 

I guess we can mention this setup. Ideally/IMO, the code-base should not be responsible for maintaining fixtures for easing documentation rendering and one must generate diagram _ad-hoc_ when needed for useful but non-essential use-cases for the project. 

Could [Brython](https://realpython.com/brython-python-in-browser/) help using `py2pulm` from a browser?

Reply:
it is really nice discussing with you, I am learning new stuffs every time. I guess you suggested to use brighton.js to compute the class diagram and display it on the main README.md? Unfortunately, github prevents from hotlinking external css and js files.

I reworded the paragraph:

```text
`py2puml` outputs diagrams in PlantUML syntax, which can be:
* versioned along your code with a unit-test ensuring its consistency (see the [test_py2puml.py's test_py2puml_model_on_py2uml_domain](tests/py2puml/test_py2puml.py) example)
* generated and hosted along other code documentation (better option: generated documentation should not be versioned with the codebase)
```